internal/services/cdn/cdn_frontdoor_rule_set_association_resource.go

```diff
@@ -0,0 +1,245 @@
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/locks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+const cdnFrontDoorRuleSetAssociationResourceType string = "azurerm_cdn_frontdoor_rule_set_association"
+
+func resourceCdnFrontDoorRuleSetAssociation() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceCdnFrontDoorRuleSetAssociationCreate,
+		Read:   resourceCdnFrontDoorRuleSetAssociationRead,
+		Update: resourceCdnFrontDoorRuleSetAssociationUpdate,
+		Delete: resourceCdnFrontDoorRuleSetAssociationDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceIdThen(func(id string) error {
+			_, err := parse.FrontDoorRuleSetAssociationID(id)
+			return err
+		}, importCdnFrontDoorRuleSetAssociation()),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"cdn_frontdoor_route_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FrontDoorRouteID,
+			},
+
+			"cdn_frontdoor_rule_set_ids": {
+				Type:     pluginsdk.TypeSet,
+				Required: true,
+
+				Elem: &pluginsdk.Schema{
+					Type:         pluginsdk.TypeString,
+					ValidateFunc: validate.FrontDoorRuleSetID,
+				},
+			},
+		},
+	}
+}
+
+func resourceCdnFrontDoorRuleSetAssociationCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	log.Printf("[INFO] preparing arguments for CDN FrontDoor Route <-> CDN FrontDoor Rule Set Association creation")
+	routeId, err := parse.FrontDoorRouteID(d.Get("cdn_frontdoor_route_id").(string))
+	if err != nil {
+		return err
+	}
+
+	// NOTE: The association name is the name of the route the resource is being associated with...
+	// e.g. subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.Cdn/profiles/profile1/afdEndpoints/endpoint1/associations/route1
+	id := parse.NewFrontDoorRuleSetAssociationID(routeId.SubscriptionId, routeId.ResourceGroup, routeId.ProfileName, routeId.AfdEndpointName, routeId.RouteName)
+
+	// lock the route and rule set resource for create...
+	locks.ByID(cdnFrontDoorRouteResourceType)
+	defer locks.UnlockByID(cdnFrontDoorRouteResourceType)
+
+	locks.ByID(cdnFrontDoorRuleSetAssociationResourceType)
+	defer locks.UnlockByID(cdnFrontDoorRuleSetAssociationResourceType)
+
+	// make sure the route and the rule sets exist and are associated with the route...
+	if err := validateRuleSetsAssociation(d, meta, &id, true); err != nil {
+		return fmt.Errorf("creating %s: %+v", id, err)
+	}
+
```

Comment:
We should likely also associate the rules here instead of depending on it done in route resource so the user doesn't have to manage two lists of rule sets

Reply:
Fixed.